src/contexts/AnalyticsContext.tsx

```diff
@@ -101,10 +106,15 @@ interface AnalyticsContextProviderProps {
 export const AnalyticsContextProvider: React.FC<
 	React.PropsWithChildren<AnalyticsContextProviderProps>
 > = ({ initialState, children }) => {
-	const [state, dispatch] = useReducer(
-		contextReducer,
-		initialState ?? initialContext,
-	);
+	const [state, dispatch] = useReducer(contextReducer, initialContext);
+	useEffect(() => {
+		if (initialState) {
+			dispatch({
+				type: AnalyticsStoreActions.setAnalyticsData,
+				payload: initialState,
+			});
+		}
+	}, [initialState]);
```

Comment:
Shouldn't this useEffect just be on mount since this is initial state? I expect that if it somehow changes at any point, we still wouldn't want to suddenly change state

Reply:
nope for mainly 2 reasons: 
-  we want to override the state in any case as initialState is of type AnalyticsContext, so if for some reason the Application that use the context wants to change it, it can.

- initialState is probably null at first render, because the flow is usually as follow:

Application first render (e.g UI) => analytics client null => initialise Analytics client => Analytics client defined